.github/workflows/pre-release.yaml

```diff
@@ -0,0 +1,64 @@
+# This workflow concerns the preparation of the `changeset` PR and keeping it updated by tracking the changes on `main` branch.
+name: Pre-Release Preparation
+
+on:
+  pull_request:
+    branches:
+      - main
+    types: [opened, reopened, labeled, synchronize]
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  pre-release:
+    name: Pre-Release Preparation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Check for changeset files and 'pre-release' label exist
+        id: check-conditions
+        run: |
+          if ls .changeset/*.md | grep '\.changeset\/[a-z-]\+\.md$' && [ "${{ contains(github.event.pull_request.labels.*.name, 'pre-release') }}" == "true" ]; then
```

Comment:
why require a pre-release label?

Reply:
Hmm, having to mark a specific PR with pre-release label seems like a weird trigger to a release. I'd rather just trigger it from an action. We can have two actions Release and Publish (like what we do now). Release action will create the PR, and publish will auto publish the release and npm package. Am I missing some benefit of your proposed approach?

Reply:
can you add a release section in the README on how to release as well? makes it easier for new contributors to understand